src/EditorFeatures/Test/Diagnostics/AnalyzerLoadFailureTests.cs

```diff
@@ -31,10 +31,7 @@ public class AnalyzerLoadFailureTests
             var eventArgs = new AnalyzerLoadFailureEventArgs(
                 errorCode,
                 message: errorCode.ToString(),
-                typeNameOpt: expectsTypeName ? analyzerTypeName : null)
-            {
-                ReferencedCompilerVersion = errorCode is AnalyzerLoadFailureEventArgs.FailureErrorCode.ReferencesNewerCompiler ? new Version(100, 0, 0, 0) : null
-            };
+                typeNameOpt: expectsTypeName ? analyzerTypeName : null);
```

Comment:
It feels like the goal of this test could be accomplished by introducing some internal helper to `CreateAnalyzerLoadFailureDiagnostic` which asserts that the `FailureErrorCode` falls within an expected range, then running the test on that helper instead of `CreateAnalyzerLoadFailureDiagnostic` itself. I'd prefer to keep the change minimal here for now, though.